ocaml/xapi/helpers.ml

```diff
@@ -150,6 +150,18 @@ let get_management_ip_addr ~__context =
   let dbg = Context.string_of_task __context in
   Option.map fst (Networking_info.get_management_ip_addr ~dbg)
 
+let get_management_interface ~__context ~host =
+  let pifs =
+    Db.PIF.get_refs_where ~__context
+      ~expr:
+        (And
+           ( Eq (Field "host", Literal (Ref.string_of host))
+           , Eq (Field "management", Literal "true")
+           )
+        )
+  in
+  match pifs with [] -> raise Not_found | pif :: _ -> pif
```

Comment:
I am not comfortable with raising `Not_found`. This function should return an option and force the caller to handle this or raise a meaningful error here.

Reply:
`List.nth_opt pifs 0` would do this